src/library_syscall.js

```diff
@@ -996,7 +997,8 @@ var SyscallsLibrary = {
       var id;
       var type;
       var name = stream.getdents.pop();
-      assert(name.length < 256); // limit of dirent struct
+      var nameLengthInBytes = lengthBytesUTF8(name);
+      assert(nameLengthInBytes < 256); // limit of dirent struct
```

Comment:
Hmm, this `assert` actually seems wrong. We can just remove it, and depend on the line `stringToUTF8(name, dirp + pos + {{{ C_STRUCTS.dirent.d_name }}}, 256);` to do its proper job with truncation.